persistence/core/src/main/scala/com/lightbend/lagom/internal/spi/projection/ProjectionSpi.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.internal.spi.projection
+
+import akka.actor.ActorSystem
+import akka.annotation.InternalStableApi
+import akka.persistence.query.EventEnvelope
+import akka.persistence.query.Offset
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+object ProjectionSpi {
+
+  @InternalStableApi
+  private[lagom] def startProcessing(system: ActorSystem, tagName: String, envelope: EventEnvelope): EventEnvelope =
+    envelope
+
+  @InternalStableApi
+  private[lagom] def afterUserFlow(projectionName: String, offset: Offset): Offset = offset
+
+  @InternalStableApi
+  private[lagom] def completedProcessing(offset: Future[Offset], exCtx: ExecutionContext): Future[Offset] = offset
```

Comment:
This seems slightly strange: is there value in capturing both the moment `completedProcessing` is called and the moment `offset` resolves? Or is there something around context propagation that means context must be captured before `offset` resolves?

Reply:
I completely missed that. Thanks @raboof